connector/countconnector/connector.go

```diff
@@ -0,0 +1,135 @@
+		}
+		setCountMetric(countScope.Metrics().AppendEmpty(), c.cfg.Traces, count, now)
+	}
+	return c.metricsConsumer.ConsumeMetrics(ctx, countMetrics)
+}
+
+func (c *count) ConsumeMetrics(ctx context.Context, md pmetric.Metrics) error {
+	now := time.Now()
+	countMetrics := pmetric.NewMetrics()
+	countMetrics.ResourceMetrics().EnsureCapacity(md.ResourceMetrics().Len())
+	for i := 0; i < md.ResourceMetrics().Len(); i++ {
+		resourceMetric := md.ResourceMetrics().At(i)
+		countResource := countMetrics.ResourceMetrics().AppendEmpty()
+		resourceMetric.Resource().Attributes().CopyTo(countResource.Resource().Attributes())
+
+		countResource.ScopeMetrics().EnsureCapacity(resourceMetric.ScopeMetrics().Len())
+		countScope := countResource.ScopeMetrics().AppendEmpty()
+		countScope.Scope().SetName(scopeName)
+
+		var count uint64
+		for j := 0; j < resourceMetric.ScopeMetrics().Len(); j++ {
+			countMetrics := resourceMetric.ScopeMetrics().At(j).Metrics()
+			for k := 0; k < countMetrics.Len(); k++ {
+				countMetric := countMetrics.At(k)
+				switch countMetric.Type() {
+				case pmetric.MetricTypeGauge:
+					count += uint64(countMetric.Gauge().DataPoints().Len())
+				case pmetric.MetricTypeSum:
+					count += uint64(countMetric.Sum().DataPoints().Len())
+				case pmetric.MetricTypeSummary:
+					count += uint64(countMetric.Summary().DataPoints().Len())
+				case pmetric.MetricTypeHistogram:
+					count += uint64(countMetric.Histogram().DataPoints().Len())
+				case pmetric.MetricTypeExponentialHistogram:
+					count += uint64(countMetric.ExponentialHistogram().DataPoints().Len())
+				}
+			}
+		}
+		setCountMetric(countScope.Metrics().AppendEmpty(), c.cfg.Metrics, count, now)
+	}
+	return c.metricsConsumer.ConsumeMetrics(ctx, countMetrics)
+}
+
+func (c *count) ConsumeLogs(ctx context.Context, ld plog.Logs) error {
+	now := time.Now()
+	countMetrics := pmetric.NewMetrics()
+	countMetrics.ResourceMetrics().EnsureCapacity(ld.ResourceLogs().Len())
+	for i := 0; i < ld.ResourceLogs().Len(); i++ {
+		resourceLog := ld.ResourceLogs().At(i)
+		countResource := countMetrics.ResourceMetrics().AppendEmpty()
+		resourceLog.Resource().Attributes().CopyTo(countResource.Resource().Attributes())
+
+		countResource.ScopeMetrics().EnsureCapacity(resourceLog.ScopeLogs().Len())
+		countScope := countResource.ScopeMetrics().AppendEmpty()
+		countScope.Scope().SetName(scopeName)
+
+		var count uint64
+		for j := 0; j < resourceLog.ScopeLogs().Len(); j++ {
+			count += uint64(resourceLog.ScopeLogs().At(j).LogRecords().Len())
+		}
+		setCountMetric(countScope.Metrics().AppendEmpty(), c.cfg.Logs, count, now)
+	}
+	return c.metricsConsumer.ConsumeMetrics(ctx, countMetrics)
+}
+
+func setCountMetric(countMetric pmetric.Metric, metricType DataTypeConfig, count uint64, now time.Time) {
+	countMetric.SetName(metricType.Name)
+	countMetric.SetDescription(metricType.Description)
+	sum := countMetric.SetEmptySum()
+	sum.SetIsMonotonic(true)
```

Comment:
May worth adding a comment here to clarify for others that is correct to be monotonic :)